frontend/js/src/tags/TagsComponent.tsx

```diff
@@ -0,0 +1,210 @@
+import { noop } from "lodash";
+import * as React from "react";
+import { useCallback, useState, useEffect } from "react";
+import {
+  ActionMeta,
+  DropdownIndicatorProps,
+  MultiValue,
+  MultiValueGenericProps,
+  components,
+} from "react-select";
+import CreatableSelect from "react-select/creatable";
+import { toast } from "react-toastify";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import TagComponent, { TagActionType } from "./TagComponent";
+import GlobalAppContext from "../utils/GlobalAppContext";
+
+type TagOptionType = {
+  value: string;
+  label: string;
+  isFixed?: boolean;
+  entityType: Entity;
+  entityMBID?: string;
+};
+
+function CreateTagText(input: string) {
+  return (
+    <>
+      <FontAwesomeIcon icon={faPlus} /> Submit new tag {input}
+    </>
+  );
+}
+
+function DropdownIndicator(props: DropdownIndicatorProps<TagOptionType>) {
+  const { isDisabled } = props;
+  if (isDisabled) {
+    return null;
+  }
+  return (
+    <components.DropdownIndicator {...props}>
+      <button className="btn btn-xs btn-outline" type="button">
+        + tag
+      </button>
+    </components.DropdownIndicator>
+  );
+}
+
+function MultiValueContainer(props: MultiValueGenericProps<TagOptionType>) {
+  const { data, selectProps } = props;
+  return (
+    <TagComponent
+      tag={{ tag: data.value, count: 1 }}
+      entityType={data.entityType}
+      entityMBID={data.entityMBID}
+      isNew={!data.isFixed}
+      deleteCallback={
+        data.isFixed
+          ? noop
+          : (tagName) => {
+              selectProps.onChange(selectProps.value, {
+                action: "remove-value",
+                removedValue: data,
+              });
+            }
+      }
+    />
+  );
+}
+function getOptionFromTag(
+  tag: ArtistTag | RecordingTag | ReleaseGroupTag,
+  entityType: Entity,
+  entityMBID?: string
+) {
+  return {
+    value: tag.tag,
+    label: tag.tag,
+    isFixed: true,
+    entityMBID: entityMBID ?? (tag as ArtistTag).artist_mbid ?? undefined,
+    entityType,
+  };
+}
+
+export default function AddTagSelect(props: {
+  entityType: "artist" | "release-group" | "recording";
+  entityMBID?: string;
+  tags?: Array<ArtistTag | RecordingTag | ReleaseGroupTag>;
+}) {
+  const { tags, entityType, entityMBID } = props;
+  const [selected, setSelected] = useState<TagOptionType[]>(
+    tags?.length
+      ? tags.map((tag) => getOptionFromTag(tag, entityType, entityMBID))
+      : []
+  );
+  useEffect(() => {
```

Comment:
This useEffect seems unnecessary.

Reply:
I see my understanding was incorrect but you can still avoid an effect it seems: https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes

Reply:
Ah yes, thank you!
I regularly forget `keys` is how it's done properly, I was scratching my head yesterday trying to figure out what I was doing wrong…